fastn-core/src/auth/mod.rs

```diff
@@ -56,8 +58,34 @@ pub async fn get_user_data_from_cookies(
 
             return match fastn_core::auth::AuthProviders::from_str(provider) {
                 fastn_core::auth::AuthProviders::GitHub => {
-                    let (user, _) =
-                        fastn_core::auth::get_authenticated_user_with_email(&session_id).await?;
+                    let user = match fastn_core::auth::get_authenticated_user_with_email(
+                        &session_id,
+                    )
+                    .await
+                    {
+                        Err(e) => {
+                            tracing::error!(
+                                "couldn't retrieve authenticated user. Reason: {:?}",
+                                e
+                            );
+
+                            if e == AuthUserError::UserDoesNotExist {
+                                return Err(fastn_core::Error::GenericError(
+                                    "User does not exist".to_string(),
+                                ));
+                            } else if let AuthUserError::UserExistsWithUnverifiedEmail(_) = e {
+                                return Err(fastn_core::Error::GenericError(
+                                    "User is not verified".to_string(),
+                                ));
+                            }
```

Comment:
These errors are not compatible with out `ftd.js`'s `http()` function.

Reply:
Write the code with the assumption that only http() function would be calling all these things. Nothing else is supported (we will never be "proper api" for outsiders to integrate).